[libpng] build failure arm64-linux shared lib Cross Compile

### Operating system

Linux (Ubuntu 22.04) on GitHub Actions

### VCPKG

2023.11.20

### Compiler

GCC

### Steps to reproduce the behavior

- Install cross compiling tools

  ```bash
    sudo apt-get install -qy binutils-aarch64-linux-gnu gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
  ```

- Install `vcpkg`

- Create custom trypletes `overlay_trypletes/arm64-linux-shared.cmake` with following content 
  
  ```cmake
     set(VCPKG_TARGET_ARCHITECTURE arm64)
     set(VCPKG_CRT_LINKAGE dynamic)
     set(VCPKG_LIBRARY_LINKAGE dynamic)

     set(VCPKG_CMAKE_SYSTEM_NAME Linux)
     set(VCPKG_BUILD_TYPE release)
  ```

- Install `libpng`

  ```bash
     vcpkg install libpng:arm64-linux-shared --overlay-triplets='./overlay_trypletes/'
  ```


### Failure logs

<details><summary>install-arm64-linux-shared-rel-out.log</summary>
<p>

Change Dir: '/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel'

Run Build Command(s): /usr/local/share/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -v -v -j5 install
[1/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DOUTPUT=pngprefix.h -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/gensrc.cmake
[2/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DOUTPUT=pnglibconf.c -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/gensrc.cmake
[3/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DOUTPUT=scripts/pnglibconf.c -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/gensrc.cmake
[4/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DINPUT=/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/scripts/symbols.c -DOUTPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/symbols.out -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/genout.cmake
[5/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DINPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/symbols.out -DOUTPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/symbols.chk -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/genchk.cmake
[6/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DINPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/pnglibconf.c -DOUTPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/pnglibconf.out -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/genout.cmake
[7/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DOUTPUT=pnglibconf.h -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/gensrc.cmake
[8/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DINPUT=/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/scripts/vers.c -DOUTPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/vers.out -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/genout.cmake
[9/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DINPUT=/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/scripts/prefix.c -DOUTPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/prefix.out -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/genout.cmake
[10/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -E remove /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/libpng.vers && /usr/local/bin/cmake -E copy /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/vers.out /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/libpng.vers
[11/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -MD -MT CMakeFiles/png-fix-itxt.dir/contrib/tools/png-fix-itxt.c.o -MF CMakeFiles/png-fix-itxt.dir/contrib/tools/png-fix-itxt.c.o.d -o CMakeFiles/png-fix-itxt.dir/contrib/tools/png-fix-itxt.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/contrib/tools/png-fix-itxt.c
[12/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DINPUT=/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/scripts/sym.c -DOUTPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/sym.out -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/genout.cmake
[13/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -E remove /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/libpng.sym && /usr/local/bin/cmake -E copy /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/sym.out /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/libpng.sym
[14/39] cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -DINPUT=/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/scripts/intprefix.c -DOUTPUT=/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/intprefix.out -P /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/scripts/genout.cmake
[15/39] : && /usr/bin/aarch64-linux-gnu-gcc -fPIC -O3 -DNDEBUG -rdynamic CMakeFiles/png-fix-itxt.dir/contrib/tools/png-fix-itxt.c.o -o png-fix-itxt  -Wl,-rpath,/usr/local/share/vcpkg/installed/arm64-linux-shared/lib  /usr/local/share/vcpkg/installed/arm64-linux-shared/lib/libz.so && :
[16/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngmem.c.o -MF CMakeFiles/png_shared.dir/pngmem.c.o.d -o CMakeFiles/png_shared.dir/pngmem.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngmem.c
[17/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngerror.c.o -MF CMakeFiles/png_shared.dir/pngerror.c.o.d -o CMakeFiles/png_shared.dir/pngerror.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngerror.c
[18/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngpread.c.o -MF CMakeFiles/png_shared.dir/pngpread.c.o.d -o CMakeFiles/png_shared.dir/pngpread.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngpread.c
[19/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngget.c.o -MF CMakeFiles/png_shared.dir/pngget.c.o.d -o CMakeFiles/png_shared.dir/pngget.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngget.c
[20/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngrio.c.o -MF CMakeFiles/png_shared.dir/pngrio.c.o.d -o CMakeFiles/png_shared.dir/pngrio.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngrio.c
[21/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngtrans.c.o -MF CMakeFiles/png_shared.dir/pngtrans.c.o.d -o CMakeFiles/png_shared.dir/pngtrans.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngtrans.c
[22/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/png.c.o -MF CMakeFiles/png_shared.dir/png.c.o.d -o CMakeFiles/png_shared.dir/png.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/png.c
[23/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngread.c.o -MF CMakeFiles/png_shared.dir/pngread.c.o.d -o CMakeFiles/png_shared.dir/pngread.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngread.c
[24/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngset.c.o -MF CMakeFiles/png_shared.dir/pngset.c.o.d -o CMakeFiles/png_shared.dir/pngset.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngset.c
[25/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngwio.c.o -MF CMakeFiles/png_shared.dir/pngwio.c.o.d -o CMakeFiles/png_shared.dir/pngwio.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngwio.c
[26/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngwrite.c.o -MF CMakeFiles/png_shared.dir/pngwrite.c.o.d -o CMakeFiles/png_shared.dir/pngwrite.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngwrite.c
[27/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/arm/arm_init.c.o -MF CMakeFiles/png_shared.dir/arm/arm_init.c.o.d -o CMakeFiles/png_shared.dir/arm/arm_init.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/arm/arm_init.c
[28/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/arm/filter_neon.S.o -MF CMakeFiles/png_shared.dir/arm/filter_neon.S.o.d -o CMakeFiles/png_shared.dir/arm/filter_neon.S.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/arm/filter_neon.S
[29/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngwtran.c.o -MF CMakeFiles/png_shared.dir/pngwtran.c.o.d -o CMakeFiles/png_shared.dir/pngwtran.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngwtran.c
[30/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngrutil.c.o -MF CMakeFiles/png_shared.dir/pngrutil.c.o.d -o CMakeFiles/png_shared.dir/pngrutil.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngrutil.c
[31/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/arm/filter_neon_intrinsics.c.o -MF CMakeFiles/png_shared.dir/arm/filter_neon_intrinsics.c.o.d -o CMakeFiles/png_shared.dir/arm/filter_neon_intrinsics.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/arm/filter_neon_intrinsics.c
[32/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/arm/palette_neon_intrinsics.c.o -MF CMakeFiles/png_shared.dir/arm/palette_neon_intrinsics.c.o.d -o CMakeFiles/png_shared.dir/arm/palette_neon_intrinsics.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/arm/palette_neon_intrinsics.c
[33/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngrtran.c.o -MF CMakeFiles/png_shared.dir/pngrtran.c.o.d -o CMakeFiles/png_shared.dir/pngrtran.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngrtran.c
[34/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -Dpng_shared_EXPORTS -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/png_shared.dir/pngwutil.c.o -MF CMakeFiles/png_shared.dir/pngwutil.c.o.d -o CMakeFiles/png_shared.dir/pngwutil.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/pngwutil.c
[35/39] : && /usr/bin/aarch64-linux-gnu-gcc -fPIC -fPIC -O3 -DNDEBUG  -Wl,--version-script='/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/libpng.vers' -shared -Wl,-soname,libpng16.so.16 -o libpng16.so.16.40.0 CMakeFiles/png_shared.dir/png.c.o CMakeFiles/png_shared.dir/pngerror.c.o CMakeFiles/png_shared.dir/pngget.c.o CMakeFiles/png_shared.dir/pngmem.c.o CMakeFiles/png_shared.dir/pngpread.c.o CMakeFiles/png_shared.dir/pngread.c.o CMakeFiles/png_shared.dir/pngrio.c.o CMakeFiles/png_shared.dir/pngrtran.c.o CMakeFiles/png_shared.dir/pngrutil.c.o CMakeFiles/png_shared.dir/pngset.c.o CMakeFiles/png_shared.dir/pngtrans.c.o CMakeFiles/png_shared.dir/pngwio.c.o CMakeFiles/png_shared.dir/pngwrite.c.o CMakeFiles/png_shared.dir/pngwtran.c.o CMakeFiles/png_shared.dir/pngwutil.c.o CMakeFiles/png_shared.dir/arm/arm_init.c.o CMakeFiles/png_shared.dir/arm/filter_neon.S.o CMakeFiles/png_shared.dir/arm/filter_neon_intrinsics.c.o CMakeFiles/png_shared.dir/arm/palette_neon_intrinsics.c.o  -Wl,-rpath,/usr/local/share/vcpkg/installed/arm64-linux-shared/lib:  /usr/local/share/vcpkg/installed/arm64-linux-shared/lib/libz.so && :
[36/39] /usr/local/bin/cmake -E cmake_symlink_library libpng16.so.16.40.0  libpng16.so.16 libpng16.so && cd /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel && /usr/local/bin/cmake -E create_symlink libpng16.so /usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel/libpng.so
[37/39] /usr/bin/aarch64-linux-gnu-gcc -DPNG_ARM_NEON_OPT=2 -I/usr/local/share/vcpkg/installed/arm64-linux-shared/include -I/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel -I/usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean -fPIC -O3 -DNDEBUG -MD -MT CMakeFiles/pngfix.dir/contrib/tools/pngfix.c.o -MF CMakeFiles/pngfix.dir/contrib/tools/pngfix.c.o.d -o CMakeFiles/pngfix.dir/contrib/tools/pngfix.c.o -c /usr/local/share/vcpkg/buildtrees/libpng/src/v1.6.40-41798b36cb.clean/contrib/tools/pngfix.c
<b style="color:red">
[38/39] : && /usr/bin/aarch64-linux-gnu-gcc -fPIC -O3 -DNDEBUG -rdynamic CMakeFiles/pngfix.dir/contrib/tools/pngfix.c.o -o pngfix  -Wl,-rpath,/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel:/usr/local/share/vcpkg/installed/arm64-linux-shared/lib  libpng16.so.16.40.0  /usr/local/share/vcpkg/installed/arm64-linux-shared/lib/libz.so && :
FAILED: pngfix 
: && /usr/bin/aarch64-linux-gnu-gcc -fPIC -O3 -DNDEBUG -rdynamic CMakeFiles/pngfix.dir/contrib/tools/pngfix.c.o -o pngfix  -Wl,-rpath,/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel:/usr/local/share/vcpkg/installed/arm64-linux-shared/lib  libpng16.so.16.40.0  /usr/local/share/vcpkg/installed/arm64-linux-shared/lib/libz.so && :
/usr/lib/gcc-cross/aarch64-linux-gnu/11/../../../../aarch64-linux-gnu/bin/ld: libpng16.so.16.40.0: undefined reference to `pow'
</b>
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

</p>
</details> 

### Additional context

The error is this:

```bash
[38/39] : && /usr/bin/aarch64-linux-gnu-gcc -fPIC -O3 -DNDEBUG -rdynamic CMakeFiles/pngfix.dir/contrib/tools/pngfix.c.o -o pngfix  -Wl,-rpath,/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel:/usr/local/share/vcpkg/installed/arm64-linux-shared/lib  libpng16.so.16.40.0  /usr/local/share/vcpkg/installed/arm64-linux-shared/lib/libz.so && :
FAILED: pngfix 
: && /usr/bin/aarch64-linux-gnu-gcc -fPIC -O3 -DNDEBUG -rdynamic CMakeFiles/pngfix.dir/contrib/tools/pngfix.c.o -o pngfix  -Wl,-rpath,/usr/local/share/vcpkg/buildtrees/libpng/arm64-linux-shared-rel:/usr/local/share/vcpkg/installed/arm64-linux-shared/lib  libpng16.so.16.40.0  /usr/local/share/vcpkg/installed/arm64-linux-shared/lib/libz.so && :
/usr/lib/gcc-cross/aarch64-linux-gnu/11/../../../../aarch64-linux-gnu/bin/ld: libpng16.so.16.40.0: undefined reference to `pow'
```

~~In [this](https://stackoverflow.com/a/12824148/20894223) post, it is indicated that to solve the issue you need to add the `-lm` flag. Can you indicate how to update `libpng` port  with this flag in the case of triplets `arm64-linux`?~~

I investigated, it doesn't depend on the `-lm` flag

--- a/ports/libpng/libm.patch
+++ b/ports/libpng/libm.patch
@@ -1,0 +1,17 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 958b1d6..21c8941 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -84,6 +84,12 @@ endif()
+ 
+ if(UNIX AND NOT APPLE AND NOT BEOS AND NOT HAIKU AND NOT EMSCRIPTEN)
+   find_library(M_LIBRARY m)
++  if(NOT M_LIBRARY)
++    find_library(M_LIBRARY m PATHS ${CMAKE_C_IMPLICIT_LINK_DIRECTORIES} NO_DEFAULT_PATH)
++    if(M_LIBRARY)
++      set(M_LIBRARY m)
++    endif()
++  endif()
+   if(NOT M_LIBRARY)
+     set(M_LIBRARY "")
+   endif()

--- a/ports/libpng/portfile.cmake
+++ b/ports/libpng/portfile.cmake
@@ -38,6 +38,7 @@ vcpkg_from_github(
         "${LIBPNG_APNG_PATCH_PATH}"
         cmake.patch
         fix-export-targets.patch
+        libm.patch
         pkgconfig.patch
         fix-msa-support-for-mips.patch
 )

--- a/ports/libpng/vcpkg.json
+++ b/ports/libpng/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "libpng",
   "version": "1.6.40",
+  "port-version": 1,
   "description": "libpng is a library implementing an interface for reading and writing PNG (Portable Network Graphics) format files",
   "homepage": "https://github.com/glennrp/libpng",
   "license": "libpng-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4686,7 +4686,7 @@
     },
     "libpng": {
       "baseline": "1.6.40",
-      "port-version": 0
+      "port-version": 1
     },
     "libpopt": {
       "baseline": "1.16",

--- a/versions/l-/libpng.json
+++ b/versions/l-/libpng.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "8b6d2c1150541a94c0ca5b2f060eb21eed4a40b2",
+      "version": "1.6.40",
+      "port-version": 1
+    },
+    {
       "git-tree": "1a2a7f489e57c9e55e049b7c5f0a28c8ff4464ed",
       "version": "1.6.40",
       "port-version": 0

